test/Dialect/FIRRTL/imconstprop.mlir

```diff
@@ -590,3 +590,28 @@ firrtl.circuit "AnnotationsBlockRemoval"  {
     firrtl.strictconnect %b, %w : !firrtl.uint<1>
   }
 }
+
+// -----
+
+firrtl.circuit "Issue3372"  {
+  firrtl.module @Issue3372(in %clock: !firrtl.clock, in %reset: !firrtl.uint<1>, out %value: !firrtl.uint<1>) {
+    %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+    %invalid_ui1 = firrtl.invalidvalue : !firrtl.uint<1>
+    %c1_ui1 = firrtl.constant 1 : !firrtl.uint<1>
+    %other_zero = firrtl.instance other interesting_name  @Other(out zero: !firrtl.uint<1>)
+    %shared = firrtl.regreset interesting_name %clock, %c0_ui1, %c1_ui1  : !firrtl.uint<1>, !firrtl.uint<1>, !firrtl.uint<1>
+    firrtl.strictconnect %shared, %shared : !firrtl.uint<1>
```

Comment:
self connect is weird here.  Is this intended?

Reply:
Yes, intended. This test is the IR of the example in https://github.com/llvm/circt/issues/3372 just before IMCP.